### PR TITLE
Add UBI image support in Makefile and Dockerfile

### PR DESCRIPTION
**Pull request by pasmant**
### Proposed changes
Describe the use case and detail of the change. If this PR addresses an issue on GitHub, make sure to include a link to that issue here in this description (not in the title of the PR).

### Checklist
Before creating a PR, run through this checklist and mark each as complete.

- [ ] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/master/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that all unit tests pass after adding my changes
- [ ] I have updated necessary documentation
- [ ] I have rebased my branch onto master
- [ ] I will ensure my PR is targeting the master branch and pulling from my branch from my own fork

### CODE REVIEW

**Review by soneillf5 — APPROVED** [review]

File: build/Dockerfile [hunk 1]
```diff
@@ -192,6 +192,45 @@ RUN --mount=type=secret,id=nginx-repo.crt,dst=/etc/ssl/nginx/nginx-repo.crt,mode
 # COPY build/*.crt  /etc/pki/ca-trust/source/anchors/
 # RUN update-ca-trust extract
 
+############################################# Base image for UBI with NGINX Plus and App Protect Dos #############################################
+FROM ubi-plus as ubi-plus-dos
+ARG NGINX_PLUS_VERSION
+
+RUN mkdir  -p /root/app_protect_dos /etc/nginx/dos/policies /etc/nginx/dos/logconfs /shared/cores /var/log/adm /var/run/adm && chmod 777 /shared/cores /var/log/adm /var/run/adm
```
Comment (lucacome) [inline build/Dockerfile:199]: This line is repeated 4 times in the Dockerfile, could you move it in the common files section?

File: build/Dockerfile [hunk 2]
```diff
@@ -192,6 +192,45 @@ RUN --mount=type=secret,id=nginx-repo.crt,dst=/etc/ssl/nginx/nginx-repo.crt,mode
...
+	--mount=type=secret,id=nginx-repo.key,dst=/etc/ssl/nginx/nginx-repo.key,mode=0644 \
+	--mount=type=secret,id=rhel_license,dst=/tmp/rhel_license,mode=0644 \
+	source /tmp/rhel_license \
+	&& subscription-manager register --org=${RHEL_ORGANIZATION} --activationkey=${RHEL_ACTIVATION_KEY} || true \
+	&& subscription-manager attach \
+    && curl -sS https://cs.nginx.com/static/files/app-protect-dos-7.repo > /etc/yum.repos.d/app-protect-dos-7.repo \
+    && subscription-manager repos --enable rhel-7-server-optional-rpms --enable rhel-7-server-extras-rpms \
+    && rpm -ivh https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm \
+    && yum clean all \
+    && yum -y install epel-release deltarpm \
```
Comment (lucacome) [inline build/Dockerfile:211]: what is `deltarpm`? I don't see it installed in NAP DOS docs.

**Review by lucacome — CHANGES_REQUESTED** [review]

File: build/Dockerfile (hunk 1 above)
Comment (pasmant) [reply build/Dockerfile:199]: moved

File: build/Dockerfile (hunk 2 above)
Comment (pasmant) [reply build/Dockerfile:211]: Removed

**Review by lucacome — APPROVED** [review]

**Pull request merged**